xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/link/DocumentXHTMLLinkTypeRenderer.java

```diff
@@ -90,6 +102,35 @@ protected String computeLabel(ResourceReference reference)
         return this.linkLabelGenerator.generate(reference);
     }
 
+    private URITitleGenerator getTitleGenerator(ResourceReference reference)
+    {
+        URITitleGenerator titleGenerator = this.defaultTitleGenerator;
+        if (this.componentManager.hasComponent(URITitleGenerator.class, reference.getType().getScheme())) {
```

Comment:
I don't see why need this as the implementation is fixed and doesn't depend on the resource reference type.